### PR TITLE
Support time series forecasting for discrete target variable

### PR DESCRIPTION
**Pull request by int-chaos**
#383 

- support 'ts_forecast_classification' task for time series forecasting where target values are categorical values
- use sklearn classifiers as learners with the help of hcrystalball sklearn wrappers
- include a test for forecasting discrete values in test/test_forecast.py 
- add 'ts_forecast_regression' as an alias for 'ts_forecast' task for time series forecasting where target values are numerical values

### CODE REVIEW

**Review by sonichi — APPROVED** [review]
Looks good to me. More documentation work to do.

File: flaml/nlp/huggingface/switch_head_auto.py [hunk 1]
```diff
@@ -5,14 +5,22 @@
...
     from transformers.models.electra.modeling_electra import ElectraClassificationHead
     from transformers.models.roberta.modeling_roberta import RobertaClassificationHead
-    from transformers.models.electra.modeling_electra import ElectraForTokenClassification
-    from transformers.models.roberta.modeling_roberta import RobertaForTokenClassification
+    from transformers.models.electra.modeling_electra import (
+        ElectraForTokenClassification,
+    )
+    from transformers.models.roberta.modeling_roberta import (
+        RobertaForTokenClassification,
+    )
```
Comment (sonichi) [inline flaml/nlp/huggingface/switch_head_auto.py:23]: merge the import statements from the same file @liususan091219

File: test/automl/test_forecast.py [hunk 2]
```diff
@@ -356,8 +356,81 @@ def test_multivariate_forecast_cat(budget=5):
...
+    print(automl.max_resource)
+    print(automl.min_resource)
+    import matplotlib.pyplot as plt
+
+    plt.title("Learning Curve")
+    plt.xlabel("Wall Clock Time (s)")
+    plt.ylabel("Validation Accuracy")
+    plt.scatter(time_history, 1 - np.array(valid_loss_history))
+    plt.step(time_history, 1 - np.array(best_valid_loss_history), where="post")
+    plt.show()
```
Comment (sonichi) [inline test/automl/test_forecast.py:428]: Does this block the test? The build fails for taking too long.

Comment (int-chaos) [reply test/automl/test_forecast.py]: I believe so since in order for it continue the plt window needs to be closed. fixed this issue in latest push

**Comment by int-chaos** [comment]
test failed for test_classification_head from test/nlp/test_autohf_classificationhead.py

**Review by slhuang — APPROVED** [review]

**Comment by sonichi** [comment]
@int-chaos, please resolve the conflict and then it's good to merge.

**Comment by firmai** [comment]
Interesting, has this been followed through?

**Pull request merged**